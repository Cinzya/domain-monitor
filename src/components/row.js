import React from 'react';
import zahnrad from "../img/zahnrad.svg";
import muelleimer from "../img/muelleimer.svg";

export const Row = (props) => (
   
    <tr key={props.id}>
        <td><img src={zahnrad} alt="Zahnradbutton für Einstellungen"/></td>
        <td id="url">{props.url}</td>
        <td><span className={props.availability}>{props.availability}</span></td>
        <td id="checked">{props.checked}</td>
        <td id="added">{props.added}</td>
        <td>
            <form >
                <input
                    id="muell"
                    className="test"
                    onClick={props.delete}
                    type="image"
                    src={muelleimer}
                    alt="Mülleimer als Button"/>
            </form>
            </td>
    </tr>
    
);
export default Row;

