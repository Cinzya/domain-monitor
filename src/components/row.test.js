import { shallow } from 'enzyme';

import React from 'react';
import ReactDOM from 'react-dom';
import {cleanup} from '@testing-library/react';
import { Row } from './row';
import deleteEvent from './list';

let container = null;

beforeEach(() => {
  // DOM element als render Ziel festlegen
  container = document.createElement("div");
  //container hängt an document dran 
  document.body.appendChild(container);
  });


afterEach(cleanup);

describe('Row', () => {
  it('expect to render Row component', () => {
    const component = shallow(<Row debug />);
    expect(component).toMatchSnapshot();
});

it('is expected to call function delete on click on the mülleimer', () => {
    const component = shallow(<Row />);
    const muelleimer = component.find('form');
    muelleimer.simulate('click');
    expect(deleteEvent).toBeCalled;
});

  
    it("renders without crashing", () => {
    ReactDOM.render(<Row/>, container)
    })
    it("should find muell image element in the DOM", () => {
        const wrapper = shallow(<Row text="test"/>)
        expect(wrapper.find("#muell")).toHaveLength(1);
    })
    it("should find muell image element with className test in the DOM",() => {
        const wrapper = shallow(<Row className="test" text="test"/>)
        expect(wrapper.find("#muell.test")).toHaveLength(1);
    })
   
      }

);

