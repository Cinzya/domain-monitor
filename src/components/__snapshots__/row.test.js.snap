// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row expect to render Row component 1`] = `ShallowWrapper {}`;
